deep_qa/training/models.py

```diff
@@ -28,6 +35,62 @@ def _summary_with_mask_info(self):
         flattened_layers = getattr(self, 'flattened_layers', self.layers)
         print_summary_with_masking(flattened_layers, getattr(self, 'container_nodes', None))
 
+    @overrides
+    def compile(self, params: Params):  # pylint: disable=arguments-differ
+        # pylint: disable=attribute-defined-outside-init
+        """
+        The only reason we are overriding this method is because keras automatically wraps
+        our tensorflow optimiser in a keras wrapper, which we don't want. We override the
+        only method in ``Model`` which uses this attribute, ``_make_train_function``, which
+        raises an error if compile is not called first.
+
+        As we move towards using a Tensorflow first optimisation loop, more things will be
+        added here which add functionality to the way Keras runs tensorflow Session calls.
+
+        """
+        optimizer = params.get('optimizer')
+        self.gradient_clipping = params.pop("gradient_clipping", None)
```

Comment:
Did this used to be inside of the optimizer params?  And now it's not, because tf does things differently?